Xamarin.Forms.Core.UnitTests/VisualTests.cs

```diff
@@ -0,0 +1,573 @@
+		}
+
+		[Test]
+		public void SetParentAndMaintainExplicitValue()
+		{
+			var layout = ExplicitMaterialLayout();
+			IVisualController view = ExplicitDefaultView();
+
+			AddExplicitDefaultToLayout(layout, (View)view);
+
+			var target = view.EffectiveVisual;
+
+			Assert.IsTrue(!target.IsMaterial(), "EffectiveVisual should be Default");
+			Assert.IsTrue(target.IsDefault(), "EffectiveVisual should be Default");
+			Assert.AreEqual(Forms.VisualMarker.Default, ((View)view).Visual);
+		}
+
+		[Test]
+		public void SetParentUsingCtorAndInheritParentValue()
+		{
+			IVisualController view = ImplicitDefaultView();
+			var layout = new StackLayout { Visual = Forms.VisualMarker.Material, Children = { (View)view } };
+
+			Assume.That(((IVisualController)layout).EffectiveVisual.IsMaterial());
+
+			Assume.That(((View)view).Visual == Forms.VisualMarker.MatchParent);
+
+			var target = view.EffectiveVisual;
+
+			Assert.IsTrue(target.IsMaterial(), "EffectiveVisual should be Material");
+			Assert.IsTrue(!target.IsDefault(), "EffectiveVisual should be Material");
+			Assert.AreEqual(Forms.VisualMarker.MatchParent, ((View)view).Visual);
+		}
+
+		[TearDown]
+		public override void TearDown()
+		{
+			base.TearDown();
+			Device.PlatformServices = null;
+		}
+
+		static void AddExplicitLTRToScrollView(ScrollView parent, View child)
+		{
+			parent.Content = child;
+
+			IVisualController controller = child;
+
+			Assume.That(controller.EffectiveVisual.IsDefault(), "child view Visual should be Default");
+			Assume.That(child.Visual == Forms.VisualMarker.Default, "child view Visual should be Default");
+		}
+
+		static void AddExplicitDefaultToLayout(StackLayout parent, View child)
+		{
+			parent.Children.Add(child);
+
+			IVisualController controller = child;
+
+			Assume.That(controller.EffectiveVisual.IsDefault(), "child view Visual should be Default");
+			Assume.That(child.Visual == Forms.VisualMarker.Default, "child view Visual should be Default");
+		}
+
+		static void AddExplicitMaterialToScrollView(ScrollView parent, View child)
+		{
+			parent.Content = child;
+
+			IVisualController controller = child;
+
+			Assume.That(child.Visual == Forms.VisualMarker.Material, "child view Visual should be Material");
+		}
+
+		static void AddExplicitMaterialToLayout(StackLayout parent, View child)
+		{
+			parent.Children.Add(child);
+
+			IVisualController controller = child;
+
+			Assume.That(controller.EffectiveVisual.IsMaterial(), "child view EffectiveVisual should be Material");
+			Assume.That(child.Visual == Forms.VisualMarker.Material, "child view Visual should be Material");
+		}
+
+		static void AddImplicitToDefault(StackLayout parent, View child)
+		{
+			parent.Children.Add(child);
+
+			IVisualController controller = child;
+
+			Assume.That(child.Visual == Forms.VisualMarker.MatchParent, "child view Visual should be MatchParent");
+		}
+
+		//static void AddImplicitToDefaultScrollView(ScrollView parent, View child)
```

Comment:
Should we just remove this?

Reply:
@samhouts 
Yea

On a side note these remained commented out because the analogous methods inside FlowDirectionTests are never called

```
AddExplicitLTRToScrollView
 AddImplicitToLTRScrollView
ExplicitLeftToRightScrollView
```

Should those be removed from the FlowDirectionTests? Or are there some tests that still should be added that use those?